app/controllers/application_controller.rb

```diff
@@ -213,6 +236,14 @@ def after_sign_in_path_for(resource)
     end
   end
 
+#destroy guest user session if a user signs out, then redirect to root path
+  def after_sign_out_path_for(resource)
+    if session[:guest_user_id]
+      session[:guest_user_id] = nil
```

Comment:
No, this just destroys the guest session; it doesn't do anything to the temporary guest account that was created in the database.  (Actually, it really just removes the "guest_user_id" from the session variable, which it what it checks for to tell that a user is a guest.)